src/Widgets/Tooltip.vala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 elementary, Inc (https://elementary.io)
+ *           2021 José Expósito <jose.exposito89@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Clutter actor to display text in a tooltip-like component.
+ *
+ * Style constants from:
+ * https://github.com/elementary/stylesheet/blob/master/src/widgets/_tooltips.scss
+ */
+public class Gala.Tooltip : Clutter.Actor {
+    private const double BACKGROUND_COLOR = (26 / 255); // #1a
+    private const double BACKGROUND_OPACITY = 0.9;
+    private const double BACKGROUND_BORDER_RADIUS = 3;
+    private const string COLOR = "#ffffff";
+    private const int PADDING_TOP = 3;
+    private const int PADDING_BOTTOM = 3;
+    private const int PADDING_LEFT = 6;
+    private const int PADDING_RIGHT = 6;
+
+    public string text {
+        get {
+            return _text;
+        }
+        set {
+            _text = value;
+            draw ();
+        }
```

Comment:
I'm generally not a fan of properties that do a lot of work when getting / setting them, because it makes it harder to follow what the code is actually doing. You may only call `draw` here but `draw` in itself creates new objects and updates the Actor's children list so it is a lot for just setting a property. Instead, these should be standard properties and the setter should decide whether to call `draw ()` or not.

Reply:
I don't like making public the draw method because then it is easy to forget calling it.
Ideally I'd like to set properties in the child text actor and avoid recreating it, but I have some issues (I'll add a comment about that in the appropriate thread).

For the moment I changed the method to a set with an optional parameter to redraw or not the tooltip so we can scratch a couple of milliseconds if required:
https://github.com/elementary/gala/pull/1076/files#diff-ed4232dda49f45599a7a209514eb953072ee8cc2515630daa9bef8719be1607cR81